dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -88,7 +88,8 @@ public TelemetryContainerFilter(final OpenTelemetry openTelemetry) {
     @Override
     public void filter(final ContainerRequestContext request) {
         Context parentContext = Context.current();
-        if ((!AgentDetection.isAgentActive()) && instrumenter.shouldStart(parentContext, request)) {
+        // instrumenter can be null if the filter isn't fully injected yet due to resource methods calls during constructor
+        if ((!AgentDetection.isAgentActive()) && instrumenter != null && instrumenter.shouldStart(parentContext, request)) {
```

Comment:
I don't understand this logic, how can `filter` possibly be called before the constructor has returned.

The only case where I could see `instrumenter` being `null` is if the no-args constructor had been called, and this should never be used except to create the CDI proxy.

There is one change we should make here, the no-args constructor on this class should be changed to package-scoped instead of `public`.